src/v/datalake/tests/record_multiplexer_bench.cc

```diff
@@ -0,0 +1,564 @@
+        co_return co_await generate_batches(
+          records_per_batch, batches, [&](auto& bb) {
+              return _record_gen.add_random_protobuf_record(
+                bb, schema_name, msg_idx, std::nullopt, gen_config);
+          });
+    }
+
+    ss::future<chunked_vector<model::record_batch>> generate_avro_batches(
+      size_t records_per_batch,
+      size_t batches,
+      std::string schema_name,
+      std::string avro_schema,
+      ::testing::avro_generator_config gen_config) {
+        co_await try_add_avro_schema(schema_name, avro_schema);
+        co_return co_await generate_batches(
+          records_per_batch, batches, [&](auto& bb) {
+              return _record_gen.add_random_avro_record(
+                bb, schema_name, std::nullopt, gen_config);
+          });
+    }
+
+    datalake::record_multiplexer create_mux() {
+        return datalake::record_multiplexer(
+          ntp,
+          std::make_unique<datalake::test_serde_parquet_writer_factory>(),
+          _schema_mgr,
+          _type_resolver,
+          _translator);
+    }
+    datalake::tests::record_generator& get_record_gen() { return _record_gen; }
+
+private:
+    std::unordered_set<std::string> _added_names;
+    datalake::catalog_schema_manager _schema_mgr;
+    datalake::record_schema_resolver _type_resolver;
+    datalake::tests::record_generator _record_gen;
+    datalake::default_translator _translator;
+};
+
+PERF_TEST_CN(
+  record_multiplexer_bench_fixture, protobuf_381_byte_message_linear_1_field) {
+    static constexpr size_t records_per_batch = 10;
+    static constexpr size_t batches = 1000;
+    static constexpr ::testing::protobuf_generator_config gen_config = {
+      .string_length_range{302, 302}};
+
+    static std::string proto_schema = generate_linear_proto(1);
+
+    static thread_local chunked_vector<model::record_batch> batch_data
+      = co_await generate_protobuf_batches(
+        records_per_batch,
+        batches,
+        "proto_schema",
+        proto_schema,
+        {0},
+        gen_config);
+
+    auto reader = model::make_fragmented_memory_record_batch_reader(
+      share_batches(batch_data));
+
+    auto consumer = counting_consumer{.mux = create_mux()};
+
+    perf_tests::start_measuring_time();
+    auto res = co_await reader.consume(std::move(consumer), model::no_timeout);
+    perf_tests::stop_measuring_time();
+
+    co_return res.total_bytes;
+}
+
+PERF_TEST_CN(
+  record_multiplexer_bench_fixture,
+  protobuf_381_byte_message_linear_40_fields) {
+    static constexpr size_t records_per_batch = 10;
+    static constexpr size_t batches = 1000;
+    static constexpr ::testing::protobuf_generator_config gen_config = {
+      .string_length_range{5, 5}};
+
+    // Generate a schema near the nesting limit.
+    // (12 payload + 3 header) bytes per field
+    static std::string proto_schema = generate_linear_proto(40);
+
+    static thread_local chunked_vector<model::record_batch> batch_data
```

Comment:
Curious, what's the significance of this being static thread_local?

Reply:
There is none, I had copied this code from an earlier iteration of the microbenchmark I had written that ran on multiple shards and it made more sense there. Will remove.

Reply:
Thanks!

Still curious, what's the significance of the remaining `static`?

Reply:
Seastar will re-run the microbenchmark multiple times to try to account for variance in the results. `static` just ensures that each run will be with the same dataset. Another, potentially better, option would be to fix the random seed.

Reply:
The other option would be to have each run be with a unique dataset. Some of the dataset characteristics like schema and string length would be constant, but the string values would be unique per-run. There's an argument for that being the better way to go. I just start with this as the default as it allows for runs to be quicker since a new dataset doesn't have to be generated each time.

Reply:
Got it, thanks for explaining.

Yeah I guess with a single data set I would expect the variance between runs to be pretty low, given there isn't a ton of non-determinism here with IO and such. Curious if you've observed that? If so, yea maybe longer term it'd be interesting to see the effects of different data sets with similarly sized data.

I don't think that needs to block this PR though

Reply:
I went ahead and tested things with random datasets on each run and variance was very low(<10ns). So I removed the static specifiers on the batch generation.